### PR TITLE
Updating libpng to v1.6.39

### PR DESCRIPTION
**Pull request by HCLJason**
I generated a new pnglibconf.h using
emconfigure ./configure
emmake make pnglibconf.h

### CODE REVIEW

**Comment by kripken** [comment]
Makes sense to me as it looks like sourceforge is the canonical place for this project. But looks like there are some CI errors.

File: tools/ports/libpng.py [hunk 1]
```diff
@@ -31,7 +31,7 @@ def get_lib_name(settings):
 
 def get(ports, settings, shared):
   # This is an emscripten-hosted mirror of the libpng repo from Sourceforge.
-  ports.fetch_project('libpng', f'https://storage.googleapis.com/webassembly/emscripten-ports/libpng-{TAG}.tar.gz', sha512hash=HASH)
+  ports.fetch_project('libpng', f'https://download.sourceforge.net/libpng/libpng-{TAG}.tar.gz', sha512hash=HASH)
```
Comment (sbc100) [inline tools/ports/libpng.py:34]: We had trouble in the past with download.sourceforge.net being unreliable (we need to to be up constantly to test this in CI).

Comment (HCLJason) [reply tools/ports/libpng.py:34]: Thanks; that's probably the source of the CI issues.
Can you please upload a copy of the libpng-1.6.39.tar.gz to the Google Storage repo? I'll revert that change.

Comment (sbc100) [reply tools/ports/libpng.py:34]: The CI issue actually looks different, I think you need to update the pngtest in the tests directory.

Comment (HCLJason) [reply tools/ports/libpng.py]: I swapped that back and fixed the pngtest test.

**Review by sbc100 — APPROVED** [review]

**Pull request merged**